src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/service/AssessmentLifecycleService.kt

```diff
@@ -22,29 +25,48 @@ class AssessmentLifecycleService {
     val log: Logger = LoggerFactory.getLogger(this::class.java)
   }
 
+  @Transactional
   fun eligibilityAnswerSubmitted(assessment: AssessmentWithEligibilityProgress): AssessmentStatus {
-    return when (assessment.calculateAggregateStatus()) {
+    return when (assessment.calculateAggregateEligibilityStatus()) {
       ELIGIBLE ->
-        // Find the previous event that occurred before eligibility checks took place
-        assessment.assessmentEntity.assessmentEvents
-          .reversed()
-          .filterIsInstance<StatusChangedEvent>()
-          .firstOrNull {
-            !setOf(
-              INELIGIBLE_OR_UNSUITABLE,
-              ELIGIBILITY_AND_SUITABILITY_IN_PROGRESS,
-              NOT_STARTED,
-            ).contains(it.changes.before)
-          }?.changes?.before ?: AssessmentStatus.ELIGIBLE_AND_SUITABLE
+        assessment.findStatusBefore(
+          INELIGIBLE_OR_UNSUITABLE,
+          ELIGIBILITY_AND_SUITABILITY_IN_PROGRESS,
+          NOT_STARTED,
+        ) ?: ELIGIBLE_AND_SUITABLE
 
       INELIGIBLE -> INELIGIBLE_OR_UNSUITABLE
       IN_PROGRESS -> ELIGIBILITY_AND_SUITABILITY_IN_PROGRESS
       else -> error("Assessment: ${assessment.assessmentEntity.id} is in an unexpected state: ${assessment.assessmentEntity.status}")
     }
   }
 
-  fun submitAssessmentForAddressChecks(assessment: AssessmentWithEligibilityProgress): AssessmentStatus = when (assessment.calculateAggregateStatus()) {
-    ELIGIBLE -> AWAITING_ADDRESS_AND_RISK_CHECKS
-    else -> error("Cannot submit an assessment that is not eligible and suitable")
+  fun submitAssessmentForAddressChecks(assessment: AssessmentWithEligibilityProgress): AssessmentStatus =
+    when (assessment.assessmentEntity.status.allowsTransitionTo(AWAITING_ADDRESS_AND_RISK_CHECKS)) {
```

Comment:
changed the logic here, rather than use eligibility info - use transitions from status to determine whether this operation is allowed